ts/components/BarcodeCamera.tsx

```diff
@@ -0,0 +1,176 @@
+};
+
+/**
+ * Map the barcode scanned through the underneath
+ * library to one of the supported ones in IO, or `null`.
+ */
+function barcodeFormatToIOFormat(
+  format: BarcodeFormat
+): IOBarcodeFormat | null {
+  switch (format) {
+    case BarcodeFormat.QR_CODE:
+      return "QRCODE";
+
+    default:
+      return null;
+  }
+}
+
+const screenWidth = Dimensions.get("screen").width;
+
+const styles = StyleSheet.create({
+  cameraContainer: {
+    position: "relative",
+    width: "100%",
+    height: screenWidth,
+    backgroundColor: "#000"
+  },
+
+  camera: {
+    position: "absolute",
+    width: "100%",
+    height: "100%"
+  },
+
+  notAuthorizedContainer: {
+    padding: customVariables.contentPadding,
+    flex: 1,
+    alignItems: "center",
+    alignSelf: "stretch",
+    marginBottom: 14
+  },
+
+  notAuthorizedText: {
+    marginBottom: 25
+  },
+
+  notAuthorizedBtn: {
+    flex: 1,
+    alignSelf: "stretch"
+  }
+});
+
+type Props = {
+  onBarcodeScanned: (barcode: ScannedBarcode) => void;
+  disabled?: boolean;
+  marker?: React.ReactNode;
+};
+
+/**
+ * A Camera view which can scan different types of
+ * barcodes (QRCodes, Data Matrix, ...).
+ */
+export const BarcodeCamera = (props: Props) => {
+  const { onBarcodeScanned, disabled } = props;
+  const devices = useCameraDevices();
+  const [permissionsGranted, setPermissionsGranted] = useState(false);
+  const device = devices.back;
+
+  const [frameProcessor, barcodes] = useScanBarcodes([BarcodeFormat.QR_CODE], {
+    checkInverted: true
+  });
+
+  // Hook that handles the permissions initialization.
+  useEffect(() => {
+    async function checkPermissions() {
+      const cameraPermissions = await Camera.getCameraPermissionStatus();
+
+      if (cameraPermissions === "not-determined") {
+        const selectedPermissions = await Camera.requestCameraPermission();
+        setPermissionsGranted(selectedPermissions === "authorized");
+      } else {
+        setPermissionsGranted(cameraPermissions === "authorized");
+      }
+    }
+
+    void checkPermissions();
+  }, [setPermissionsGranted]);
+
+  // Hook that handles the `onBarcodeScanned` callback.
+  useEffect(() => {
+    // This is going to take only the first scanned
+    // barcode. This could be improved or changed in relation
+    // to the business decisions.
+    const firstBarcode = barcodes[0];
+
+    if (!firstBarcode) {
+      return;
+    }
+
+    const barcodeValue = firstBarcode.content.data.toString();
+    const barcodeFormat = barcodeFormatToIOFormat(firstBarcode.format);
+
+    if (barcodeFormat === null) {
+      return;
+    }
+
+    // Execute an haptic feedback
+    Vibration.vibrate(VIBRATION_BARCODE_SCANNED_DURATION);
+
+    onBarcodeScanned({
+      format: barcodeFormat,
+      value: barcodeValue
+    });
+  }, [barcodes, onBarcodeScanned]);
+
+  if (!permissionsGranted) {
+    return (
+      <View style={styles.notAuthorizedContainer}>
+        <Text style={styles.notAuthorizedText}>
```

Comment:
You're right! I used the `Body` component because it reflected more the old text used there in the previous implementation (this is the commit: https://github.com/pagopa/io-app/pull/3928/commits/a644fd9360686dc5b381ee52f89a868443cb1165). What do you think?

<img src="https://user-images.githubusercontent.com/5963683/167626030-49755f67-5f4c-453a-91cf-9a665cdfcc19.jpeg" height="350" />

Reply:
Sorry I left a vague indication 
You solution is the best one